x/staking/handler.go

```diff
@@ -220,7 +221,8 @@ func handleMsgDelegate(ctx sdk.Context, msg types.MsgDelegate, k keeper.Keeper)
 		return ErrBadDenom(k.Codespace()).Result()
 	}
 
-	_, err := k.Delegate(ctx, msg.DelegatorAddress, msg.Amount.Amount, validator, true)
+	// NOTE source will always be from a wallet which are unbonded
```

Comment:
Really, what this should say is:

```suggestion
	// NOTE: source funds are always unbonded
```

This is because I, suppose I'm a validator, further delegate to myself. The source (me as a validator) is bonded, but the funds I'm bonding are unbonded.